src/applications/apps.js

```diff
@@ -25,6 +24,50 @@ import { assign } from "../utils/assign";
 
 const apps = [];
 
+export function getAppChanges() {
+  const appsToUnload = [],
+    appsToUnmount = [],
+    appsToLoad = [],
+    appsToMount = [];
+
+  // We re-attempt to download applications in LOAD_ERROR after a timeout of 200 milliseconds
+  const currentTime = new Date().getTime();
+
+  apps.forEach((app) => {
+    const appShouldBeActive =
+      app.status !== SKIP_BECAUSE_BROKEN && shouldBeActive(app);
+
+    switch (app.status) {
+      case LOAD_ERROR:
+        if (currentTime - app.loadErrorTime >= 200) {
+          appsToLoad.push(app);
+        }
+        break;
+      case NOT_LOADED:
+        if (appShouldBeActive) {
+          appsToLoad.push(app);
+        }
+        break;
+      case NOT_BOOTSTRAPPED:
+      case NOT_MOUNTED:
+        if (!appShouldBeActive && getAppUnloadInfo(toName(app))) {
+          appsToUnload.push(app);
+        } else if (appShouldBeActive) {
+          appsToMount.push(app);
+        }
+        break;
+      case MOUNTED:
+        if (!appShouldBeActive) {
+          appsToUnmount.push(app);
+        }
+        break;
+      // all other statuses are ignored
+    }
+  });
+
+  return [appsToUnload, appsToUnmount, appsToLoad, appsToMount];
```

Comment:
this is minor but I'm not sure I like this vs just have an object with named properties.

this pattern is good if you don't want to enforce a naming convention on the users that consume it (e.g. why it works great for react hooks) but the downside is that you have to memorize the order of data that is returned. 

in this case, it seems like it shouldn't be a big deal that we enforce what these are named, and then consumers don't have to memorize the order (and don't have to do empty array destructuring if, say, they only want `appsToLoad` but nothing else)

Reply:
I agree that it's easier to use as an object, but Guy Bedford has corrupted me and I now think about bytes. If the function returns named properties on an object they can't usually be mangled, whereas array element names can be. Since this function wasn't exposed as a public API and only used in one place, I figured it was okay to get the order right. 

However, I just compared the byte savings and it turns out I was completely wrong and using the object properties is actually a lot less bytes? around 900 fewer bytes. I have no idea why, but am going to go with it.

Reply:
that's weird. wonder if it has to do with how they're transpiled? 🤷‍♀️

Reply:
yeah I guess so. I was crunched for time so i didn't look at the output bundles, but my guess would be something related to destructuring arrays vs destructuring objects?